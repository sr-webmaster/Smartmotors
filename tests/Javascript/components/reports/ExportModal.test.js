import { mount } from '@vue/test-utils'
import createLocalVue from 'vue-cli-plugin-freshinup-ui/utils/testing/createLocalVue'
import Component from '~/components/reports/ExportModal.vue'

describe('Export Modal', () => {
  let localVue

  beforeEach(() => {
    const vue = createLocalVue()
    localVue = vue.localVue
  })

  describe('Methods', () => {
    test('exportReport emits export event with payload', () => {
      const wrapper = mount(Component, { localVue })

      wrapper.setData({
        option: 'usque'
      })

      wrapper.vm.exportReport()
      expect(wrapper.emitted().export).toHaveLength(1)
      expect(wrapper.emitted().export[0][0]).toEqual('usque')
    })
  })

  describe('Visuals', () => {
    test('default', () => {
      const wrapper = mount(Component, { localVue })

      expect(wrapper.find('.v-dialog').isVisible()).toBe(false)
      expect(wrapper.element).toMatchSnapshot()
    })

    test('click on export button', async () => {
      const wrapper = mount(Component, { localVue })

      wrapper.find('.sm-ReportsExportModal__activator').trigger('click')
      await wrapper.vm.$nextTick()
      expect(wrapper.find('.v-dialog').isVisible()).toBe(true)
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
