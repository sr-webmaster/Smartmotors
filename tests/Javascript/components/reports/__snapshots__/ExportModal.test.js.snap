// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Export Modal Visuals click on export button 1`] = `
<div
  class="v-dialog__container"
  style="display: inline-block;"
>
  <button
    class="sm-ReportsExportModal__activator v-btn v-btn--round theme--light primary"
    type="button"
  >
    <div
      class="v-btn__content"
    >
      
      Export
    
    </div>
  </button>
  <div
    class="v-dialog__content v-dialog__content--active"
    style="z-index: 202; z-index: 202;"
    tabindex="-1"
  >
    <div
      class="v-dialog v-dialog--active dialog-transition-enter dialog-transition-enter-active"
      style="max-width: 400px;"
    >
       
      <div
        class="v-card v-sheet theme--light"
      >
        <nav
          class="v-toolbar theme--light"
          style="margin-top: 0px; padding-right: 0px; padding-left: 0px; transform: translateY(0px);"
        >
          <div
            class="v-toolbar__content"
            style="height: 64px;"
          >
            <div
              class="v-toolbar__title"
            >
              Export Report
            </div>
             
            <div
              class="spacer"
            />
             
            <div
              class="v-toolbar__items"
            >
              <button
                class="v-btn v-btn--icon theme--light"
                type="button"
              >
                <div
                  class="v-btn__content"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon material-icons theme--light"
                  >
                    close
                  </i>
                </div>
              </button>
            </div>
          </div>
        </nav>
         
        <div
          class="v-card__text"
        >
          <form
            class="v-form"
            novalidate="novalidate"
          >
            <div
              class="v-input v-input--selection-controls v-input--radio-group v-input--radio-group--column v-input--is-label-active v-input--is-dirty theme--light"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                  style="height: auto;"
                >
                  <div
                    class="v-input--radio-group__input"
                    role="radiogroup"
                  >
                    <div
                      class="v-radio theme--light"
                    >
                      <div
                        class="v-input--selection-controls__input"
                      >
                        <input
                          aria-checked="false"
                          aria-label="PDF"
                          name="v-radio-84"
                          role="radio"
                          type="radio"
                          value="pdf"
                        />
                        <div
                          class="v-input--selection-controls__ripple"
                        />
                        <i
                          aria-hidden="true"
                          class="v-icon material-icons theme--light"
                        >
                          radio_button_unchecked
                        </i>
                      </div>
                      <label
                        aria-hidden="true"
                        class="v-label theme--light"
                        style="left: 0px; position: relative;"
                      >
                        PDF
                      </label>
                    </div>
                     
                    <div
                      class="v-radio theme--light accent--text"
                    >
                      <div
                        class="v-input--selection-controls__input"
                      >
                        <input
                          aria-checked="true"
                          aria-label="CSV"
                          name="v-radio-84"
                          role="radio"
                          type="radio"
                          value="csv"
                        />
                        <div
                          class="v-input--selection-controls__ripple accent--text"
                        />
                        <i
                          aria-hidden="true"
                          class="v-icon material-icons theme--light accent--text"
                        >
                          radio_button_checked
                        </i>
                      </div>
                      <label
                        aria-hidden="true"
                        class="v-label theme--light"
                        style="left: 0px; position: relative;"
                      >
                        CSV
                      </label>
                    </div>
                  </div>
                </div>
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </form>
        </div>
         
        <div
          class="v-card__actions"
        >
          <button
            class="v-btn v-btn--block v-btn--flat theme--light secondary--text"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
        Cancel
      
            </div>
          </button>
           
          <button
            class="v-btn v-btn--block theme--light primary"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
        Export
      
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Export Modal Visuals default 1`] = `
<div
  class="v-dialog__container"
  style="display: inline-block;"
>
  <button
    class="sm-ReportsExportModal__activator v-btn v-btn--round theme--light primary"
    type="button"
  >
    <div
      class="v-btn__content"
    >
      
      Export
    
    </div>
  </button>
  <div
    class="v-dialog__content"
    style="z-index: 0;"
    tabindex="-1"
  >
    <div
      class="v-dialog"
      style="max-width: 400px; display: none;"
    >
       
      <div
        class="v-card v-sheet theme--light"
      >
        <nav
          class="v-toolbar theme--light"
          style="margin-top: 0px; padding-right: 0px; padding-left: 0px; transform: translateY(0px);"
        >
          <div
            class="v-toolbar__content"
            style="height: 64px;"
          >
            <div
              class="v-toolbar__title"
            >
              Export Report
            </div>
             
            <div
              class="spacer"
            />
             
            <div
              class="v-toolbar__items"
            >
              <button
                class="v-btn v-btn--icon theme--light"
                type="button"
              >
                <div
                  class="v-btn__content"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon material-icons theme--light"
                  >
                    close
                  </i>
                </div>
              </button>
            </div>
          </div>
        </nav>
         
        <div
          class="v-card__text"
        >
          <form
            class="v-form"
            novalidate="novalidate"
          >
            <div
              class="v-input v-input--selection-controls v-input--radio-group v-input--radio-group--column v-input--is-label-active v-input--is-dirty theme--light"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                  style="height: auto;"
                >
                  <div
                    class="v-input--radio-group__input"
                    role="radiogroup"
                  >
                    <div
                      class="v-radio theme--light"
                    >
                      <div
                        class="v-input--selection-controls__input"
                      >
                        <input
                          aria-checked="false"
                          aria-label="PDF"
                          name="v-radio-48"
                          role="radio"
                          type="radio"
                          value="pdf"
                        />
                        <div
                          class="v-input--selection-controls__ripple"
                        />
                        <i
                          aria-hidden="true"
                          class="v-icon material-icons theme--light"
                        >
                          radio_button_unchecked
                        </i>
                      </div>
                      <label
                        aria-hidden="true"
                        class="v-label theme--light"
                        style="left: 0px; position: relative;"
                      >
                        PDF
                      </label>
                    </div>
                     
                    <div
                      class="v-radio theme--light"
                    >
                      <div
                        class="v-input--selection-controls__input"
                      >
                        <input
                          aria-checked="false"
                          aria-label="CSV"
                          name="v-radio-48"
                          role="radio"
                          type="radio"
                          value="csv"
                        />
                        <div
                          class="v-input--selection-controls__ripple"
                        />
                        <i
                          aria-hidden="true"
                          class="v-icon material-icons theme--light"
                        >
                          radio_button_unchecked
                        </i>
                      </div>
                      <label
                        aria-hidden="true"
                        class="v-label theme--light"
                        style="left: 0px; position: relative;"
                      >
                        CSV
                      </label>
                    </div>
                  </div>
                </div>
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </form>
        </div>
         
        <div
          class="v-card__actions"
        >
          <button
            class="v-btn v-btn--block v-btn--flat theme--light secondary--text"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
        Cancel
      
            </div>
          </button>
           
          <button
            class="v-btn v-btn--block theme--light primary"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
        Export
      
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
